lib/streamlit/delta_generator.py

```diff
@@ -487,6 +487,10 @@ def _enqueue(
                 cursor=new_cursor,
                 parent=dg,
             )
+
+            # Elements inherit their parent form ids.
+            # NOTE: Form ids aren't set in dg constructor.
+            output_dg._form_data = FormData(current_form_id(dg))
```

Comment:
It would be possible to accept `_form_data` as an argument for dg construction and pass it at `output_dg` creation on line 485, right?